### PR TITLE
implement redirect

### PR DESCRIPTION
**Pull request by junglie85**
Adds redirect capability and tests for a couple of missing controller cases.

### CODE REVIEW

**Comment by junglie85** [comment]
@jsuchy please could you review? All acceptance tests pass. I do however need to do some work on the packaging structure, which I'll probably leave for another PR.

File: src/test/java/uk/ashleybye/httpserver/server/ServerTest.java [hunk 1]
```diff
@@ -167,6 +210,17 @@ void testPostMethod() {
     assertEquals("HTTP/1.1 200 OK\n\r\nsome body", connection.getSentData());
   }
 
+  @Test
+  void testRedirect() {
```
Comment (cmvandrevala) [inline src/test/java/uk/ashleybye/httpserver/server/ServerTest.java]: I would recommend using more descriptive test names here. For example, "testRedirectReturnsTheCorrectHttpHeaders" or something like that

File: src/main/java/uk/ashleybye/httpserver/http/controller/MethodOptionsController.java [hunk 2]
```diff
@@ -1,17 +1,10 @@
...
 
 import uk.ashleybye.httpserver.http.RequestMethod;
-import uk.ashleybye.httpserver.server.Request;
-import uk.ashleybye.httpserver.server.Response;
 
 public class MethodOptionsController extends Controller {
 
   public MethodOptionsController(RequestMethod... allowedMethods) {
     super(allowedMethods);
   }
```
Comment (cmvandrevala) [inline src/main/java/uk/ashleybye/httpserver/http/controller/MethodOptionsController.java:9]: Nice. I like this code cleanup here.

File: .travis.yml [hunk 3]
```diff
@@ -9,6 +9,6 @@ script:
   - ./gradlew test jacocoTestReport
   - ./gradlew jar
   - java -jar build/http-server.jar &
-  - cd acceptance-tests && bundle install && bundle exec spinach -t @simple-get,@not-found,@simple-head,@simple-options,@not-allowed,@simple-post && cd ..
+  - cd acceptance-tests && bundle install && bundle exec spinach && cd ..
```
Comment (cmvandrevala) [inline .travis.yml]: Is kebab case the standard in Java projects? Shouldn't this be camel case?

File: src/main/java/uk/ashleybye/httpserver/http/StatusCode.java [hunk 4]
```diff
@@ -37,6 +37,18 @@ public String getStatusMessage() {
     }
   },
 
+  MOVED_PERMANENTLY {
```
Comment (cmvandrevala) [inline src/main/java/uk/ashleybye/httpserver/http/StatusCode.java:40]: This structure will certainly work to organize status codes and messages, but it does seem a bit verbose. Is there a more concise way to store this information?

**Pull request merged**